frappe/public/js/frappe/form/footer/form_timeline.js

```diff
@@ -607,31 +607,26 @@ class FormTimeline extends BaseTimeline {
 			});
 	}
 
+	/**
+	 * Return the latest email communication.
+	 *
+	 * @param {boolean} from_recipient If true, only considers emails where the recipient is in the list of senders.
+	 * @returns {object|null} The latest email communication, or null if no communication is found.
+	 */
 	get_last_email(from_recipient) {
-		let last_email = null;
-		let communications = this.frm.get_docinfo().communications || [];
-		let email = this.get_recipient();
-		// REDESIGN TODO: What is this? Check again
-		communications
-			.sort((a, b) => (a.creation > b.creation ? -1 : 1))
-			.forEach((c) => {
-				if (
-					c.communication_type === "Communication" &&
-					c.communication_medium === "Email"
-				) {
-					if (from_recipient) {
-						if (c.sender.indexOf(email) !== -1) {
-							last_email = c;
-							return false;
-						}
-					} else {
-						last_email = c;
-						return false;
-					}
-				}
-			});
-
-		return last_email;
+		const communications = this.frm.get_docinfo().communications || [];
+		const email = this.get_recipient();
+
+		const filteredRecords = communications
+			.filter(
+				(record) =>
+					record.communication_type === "Communication" &&
+					record.communication_medium === "Email" &&
+					(!from_recipient || record.sender.indexOf(email) !== -1)
```

Comment:
```suggestion
					(!from_recipient || record.sender === email)
```

Based on my data, this should work just as well and be more readable.

Reply:
Yeah, since `record.sender` is a parsed email and there can't be more than one sender, comparing like this should have the same effect.

Ideally we should be `casefold`-ing and stripping the emails before comparing.